### PR TITLE
Centralized generator of error messages

### PR DESCRIPTION
**Pull request by rlnx**
Current:
- Introduced `error_messages` class with the static functions returning `const char *`  for all possible error messages, name of the function reflects content of the message, but not tightly coupled with it to have flexibility in changing error message content.
-  Replace all raw string in throw expressions across the oneAPI part

Future:
- Support [messages formatting](https://en.cppreference.com/w/cpp/utility/format)
- Generate `error_messages` and `exception_generator` from YAML:
```yaml
input_data_is_empty: >
   Input data is empty

input_labels_are_empty: >
   Input labels are empty
```

### CODE REVIEW

File: cpp/oneapi/dal/algo/svm/detail/infer_ops.hpp [hunk 1]
```diff
@@ -42,51 +42,46 @@ struct infer_ops {
...
 
     void check_preconditions(const Descriptor& params, const input_t& input) const {
-        if (!(input.get_data().has_data())) {
-            throw domain_error("Input data should not be empty");
-        }
-        if (input.get_model().get_support_vector_count() < 0) {
-            throw invalid_argument("Input model support_vector_count should be >= 0");
+        using msg = dal::detail::error_messages;
+
+        // TODO: Should be invariant of model
```
Comment (PetrovKP) [inline cpp/oneapi/dal/algo/svm/detail/infer_ops.hpp:47]: do you mean that this should be checked in the model and in the kernel functions?

**Review by PetrovKP — APPROVED** [review]
Cool changes, thanks! For kernel functions and svm I looked, it looks good.

File: cpp/oneapi/dal/detail/error_messages.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    "Values of bootstrap and error metric parameters provided "
+    "in descriptor are incompatible to each other")
+MSG(bootstrap_is_incompatible_with_variable_importance_mode,
+    "Values of bootstrap and variable importance mode parameters provided "
+    "in descriptor are incompatible to each other")
+MSG(decision_forest_train_dense_method_is_not_implemented_for_gpu,
+    "Decision forest train dense method is not implemented for GPU")
+MSG(decision_forest_train_hist_method_is_not_implemented_for_cpu,
+    "Decision forest train hist method is not implemented for CPU")
+
```
Comment (mchernov-intel) [inline cpp/oneapi/dal/detail/error_messages.cpp:155]: `#undef MSG` doesn't matter cause it's cpp file?

File: cpp/oneapi/dal/backend/linalg/matrix.hpp [hunk 3]
```diff
@@ -143,7 +143,7 @@ class matrix : public matrix_base {
     static matrix wrap(const table& t, layout l = layout::row_major) {
         if (l != layout::row_major) {
             // TODO: Figure out how to use column-major layout
-            throw unimplemented{ "Only row-major is supported" };
+            throw unimplemented{ dal::detail::error_messages::only_row - major_is_supported() };
```
Comment (mchernov-intel) [inline cpp/oneapi/dal/backend/linalg/matrix.hpp:146]: asked offline

**Review by mchernov-intel — APPROVED** [review]

File: cpp/oneapi/dal/detail/error_messages.cpp (hunk 2 above)
Comment (rlnx) [reply cpp/oneapi/dal/detail/error_messages.cpp:155]: yes

**Comment by rlnx** [comment]
@PetrovKP, @mchernov-intel  I'll reopen PR and rename branch, private CI treats the name of this branch `central-error-messages-2` as an error 😃

**Pull request closed without merging**